src/pipeline.ts

```diff
@@ -489,14 +551,483 @@ export async function scanRssFeeds() {
+		INJECTED: 0,
+		FULL_MATCHES: 0,
+		PARTIAL_MATCHES: 0,
+		BLOCKED: 0,
+		ALREADY_EXISTS: 0,
+		INCOMPLETE_CANDIDATES: 0,
+		INCOMPLETE_SEARCHEES: 0,
+		FAILED: 0,
+		UNMATCHED: 0,
+		FOUND_BAD_FORMAT: false,
+		FLAT_LINKING: flatLinking,
+	};
+	let count = 0;
+	for (const torrentFilePath of dirContents) {
+		const progress = chalk.blue(`(${++count}/${dirContents.length})`);
+		const filePathLog = getTorrentFilePathLog(torrentFilePath);
+		if (shouldCleanUpTorrent(torrentFilePath)) {
+			toDelete.add(torrentFilePath);
+		}
+		let meta: Metafile;
+		try {
+			meta = await parseTorrentFromFilename(torrentFilePath);
+		} catch (e) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to parse ${filePathLog}`,
+			});
+			logger.debug(e);
+			continue;
+		}
+		const metaLog = getLogString(meta, chalk.bold.white);
+
+		const torrentNameInfo =
+			await parseInfoFromSavedTorrent(torrentFilePath);
+		const tracker = torrentNameInfo?.tracker ?? UNKNOWN_TRACKER;
+		if (tracker === UNKNOWN_TRACKER) {
+			summary.FOUND_BAD_FORMAT = true;
+		}
+
+		const { fullMatches, partialMatches, foundBlocked } =
+			await injectDecideStage(meta, searchees);
+		if (fullMatches.length === 0 && partialMatches.length === 0) {
+			if (foundBlocked) {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} ${metaLog} ${chalk.yellow("possibly blocklisted")}: ${filePathLog}`,
+				});
+				summary.BLOCKED++;
+			} else {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} ${metaLog} ${chalk.red("has no matches")}: ${filePathLog}`,
+				});
+				summary.UNMATCHED++;
+			}
+			continue;
+		}
+
+		const {
+			injectionResult,
+			matchedSearchee,
+			matchedDecision,
+			linkedNewFiles,
+		} = await injectInitialAction(
+			meta,
+			fullMatches,
+			partialMatches,
+			tracker,
+		);
+		if (injectionResult === InjectionResult.FAILURE) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to inject ${filePathLog} - ${chalk.red(injectionResult)}`,
+			});
+			summary.FAILED++;
+			toDelete.delete(torrentFilePath);
+			continue;
+		}
+		if (injectionResult === InjectionResult.TORRENT_NOT_COMPLETE) {
+			if (
+				!linkDir ||
+				fs.statSync(torrentFilePath).mtimeMs >= Date.now() - ms("1 day")
+			) {
+				// Normal case where source is likely still downloading
+				logger.warn({
+					label: Label.INJECT,
+					message: `${progress} Unable to inject ${filePathLog} - ${chalk.yellow(injectionResult)}`,
+				});
+			} else {
+				// Since source is stalled, add to client paused so user can resume later if desired
+				// Try linking all possible matches as they may have different files
+				await injectFromStalledTorrent(
+					meta,
+					fullMatches,
+					partialMatches,
+					tracker,
+					injectionResult,
+					progress,
+					filePathLog,
+				);
+			}
+			summary.INCOMPLETE_SEARCHEES++;
+			continue;
+		}
+		const result = await getClient().isTorrentComplete(meta.infoHash);
```

Comment:
is there a reason we can't trust the `injectionResult` on hand and have to call the client again to check for completion?

Reply:
If the torrent already exists, it's used for logging and behavior.